### DISCUSSION TITLE
Idea: Make outline awesome for opensource development. Use Passport JS for authentication

### DISCUSSION THREAD

**Discussion by chsasank** [outline/outline#1531] [General]
Github wiki is quite bad. Using outline is far superior experience than github wiki for opensource development.

To make outline *the* wiki for opensource development, we just have to do the following

* No login at all for read access
* Login with github and access control for write access or private collections

That's it, outline will be preferred wiki tool for opensource development :)

I already do KB for my project on outline. But people need to be on slack for giving the access.

I understand that this is not the first request for logins. How about we support general login mechanism and easy extensibility with technologies like

* SAML
* OAuth
* [Passport js](http://www.passportjs.org/) to implement it all!

Once we get passportjs in, authentication should become quite easy overall.

**Comment by matbrgz** [comment]
You could help to finish this https://github.com/outline/outline/pull/1183

**Comment by almereyda** [comment]
With Passport landing some time soon'ish, I also wonder how nice Outline would be as a public knowledge base like @documize

Are there any plans to support publicly readable pages via canonical URLs?

**Reply by tommoor** [reply]
We haven't put a lot of thought into publicly exposing the canonical URL's to be honest, I'm certainly not ruling it out – but there are a lot of OSS options for publicly hosted wiki's already. I think that internal knowledge bases are underserved in comparison and it's probably best for us to continue focusing efforts there.

**Reply by chsasank** [reply]
We should do it if and only if it's not a big effort to exposing things publicly.
